src/Features/Core/Portable/AddConstructorParametersFromMembers/AddConstructorParametersFromMembersCodeRefactoringProvider.cs

```diff
@@ -151,5 +154,37 @@ static CodeAction GetOptionalContructorParametersCodeAction(Document document, C
                     document, constructorCandidate, containingType, missingOptionalParameters, useSubMenuName);
             }
         }
+
+        public async Task<ImmutableArray<IntentProcessorResult>> ComputeIntentAsync(Document priorDocument, TextSpan priorSelection, Document currentDocument, string serializedIntentData, CancellationToken cancellationToken)
+        {
+            var actions = await AddConstructorParametersFromMembersAsync(priorDocument, priorSelection, cancellationToken).ConfigureAwait(false);
+            if (actions.IsEmpty)
+            {
+                return ImmutableArray<IntentProcessorResult>.Empty;
+            }
+
+            // This provider generates nested actions, so we need to flatten them before we compute the edits.
+            actions = actions.SelectMany(a => GetNestedActions(a)).ToImmutableArray();
```

Comment:
hrmm... could we instead haveAddConstructorParametersFromMembersAsync return a richer structure that is easier for you to extract information from here, and then the main refactoring path package tahts into CodeActions?

Reply:
Updated, let me know what you think!